EntityFramework.Functions/Function.DbModel.cs

```diff
@@ -557,12 +557,34 @@ private static IList<FunctionParameter> GetStoreParameters
             this DbModel model, MethodInfo methodInfo)
         {
             ParameterInfo[] parameters = methodInfo.GetParameters().ToArray();
-            return parameters
-                .Select((parameterInfo) => FunctionParameter.Create(
-                    parameterInfo.GetCustomAttribute<ParameterAttribute>()?.Name ?? parameterInfo.Name,
-                    model.GetModelStructualType(parameterInfo.ParameterType, methodInfo),
-                    ParameterMode.In))
-                .ToArray();
+            return parameters.Select(
+                (parameterInfo) =>
+                {
+                    EdmType type = model.GetModelPrimitiveType(parameterInfo.ParameterType, methodInfo);
+                    if (type == null)
+                    {
+                        type = model.GetModelComplexType(parameterInfo.ParameterType, methodInfo);
+                        if (type == null)
+                        {
+                            type = model.GetModelEntityType(parameterInfo.ParameterType, methodInfo);
+                            if (type == null)
+                            {
+                                type = model.GetModelStructualType(parameterInfo.ParameterType, methodInfo);
```

Comment:
model.GetModelStructualType calls model.GetModelComplexType and model.GetModelEntityType internally, and throw NotSupportedException.